Release 0.1 (stable)

woju opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/intel/ScaffoldingForGramine/22)
<!-- Reviewable:end -->

oshogbo committed:
Add numeric representation of framework options (#2)

In interactive mode, enable the selection of framework by either providing its
name as a string or its sequence number.

Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>

oshogbo committed:
Add an option to bootstrap a project (#3)

Bootstrapping is the straightforward process of generating example
files from a specified framework.

Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>

oshogbo committed:
Add nodejs framework

Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>

oshogbo committed:
docker: update package repository (#4)

Ensure package information is up-to-date before installing new
software.

Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>

oshogbo committed:
Fix zuul playbook

Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>

oshogbo committed:
python_plain: Enable remote DCAP attestation (#5)

As we are here, allow to add new trusted_files.

Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>

oshogbo committed:
Add framework specific help (#7)

Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>

oshogbo committed:
Fix project name

Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>

woju committed:
README,debian: Fix installation instruction to install docker.io

Also remind to install gramine for Development setup in README

Signed-off-by: Wojtek Porczyk <woju@invisiblethingslab.com>

oshogbo committed:
Fix setup (#10)

Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>

oshogbo committed:
Add koajs framework

Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>
Co-authored-by: Wojtek Porczyk <woju@invisiblethingslab.com>
Signed-off-by: Wojtek Porczyk <woju@invisiblethingslab.com>

oshogbo committed:
Add `and-run` option to the build process.

This option allows users to avoid manually copying the docker-id,
streamlining the build and run process. It's particularly useful
during the development of new apps where frequent builds and runs
are common.

Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>

oshogbo committed:
Add expressjs framework

Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>
Co-authored-by: Wojtek Porczyk <woju@invisiblethingslab.com>
Signed-off-by: Wojtek Porczyk <woju@invisiblethingslab.com>

omeg committed:
Add dotnet framework

and "dotnet: build the application in docker"
and "Add dotnet framework to CI"

Signed-off-by: Rafał Wojdyła <omeg@invisiblethingslab.com>
Co-authored-by: Wojtek Porczyk <woju@invisiblethingslab.com>
Signed-off-by: Wojtek Porczyk <woju@invisiblethingslab.com>

oshogbo committed:
Fix names of directories and jobs

Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>

oshogbo committed:
All tests require aesm

Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>

oshogbo committed:
Consolidate apt-get commands and resolve hadolint warnings

This commit consolidates two apt-get commands to ensure that
the `apt-get update` command is synchronized with
`apt-get install`. This prevents potential issues where Docker
caches the layer of each RUN command, potentially using outdated
distribution files with `apt-get install`.

Additionally, this commit addresses hadolint warnings.

Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>

oshogbo committed:
Define the image variable in the global scope

Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>

oshogbo committed:
Delete the apt-get lists after installing something

Also add this check as ignored by hadolint.
Hadolint requires after each apt-get install this line,
but I don't want to force user to remember about that.
We have this in main Dockerfile, which means that it will
be added to all Dockerfiles automagically.

Requested by hadolint.

Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>

oshogbo committed:
Ignore Multiple RUN Cases

We are keeping commands in separate RUN statements to avoid
requiring the user to manually run them, as our Dockerfile
is seperated in multiple sections.

Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>

oshogbo committed:
Extend Expect timeout to 90 seconds

The default timeout for Expect is 10 seconds, which may not suffice
as the Docker project build time can exceed this duration.
To accommodate longer build times, the timeout has been extended to
90 seconds.

Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>

oshogbo committed:
ci: make DL3059 hadolint rule ignored globally

This Rule catches multiple consecutive RUN instructions. Seems
that we can't avoid this issue.

Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>

oshogbo committed:
Add information about SGX signing key

Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>

woju committed:
Build using mmdebstrap

- Builds in chroot (not in docker), and allows to run gramine-sgx-sign
  from outside of the chroot (because of plugins); the catch is that
  the two packages need to match bit for bit.
- Requires Gramine 1.6 (not yet released as of time of this writing).
- For now, until Gramine 1.6 is released, use unstable-* repos.
- Update Ubuntu 20.04 (jammy) to Debian 12 (bookworm).
- Because of that, few things shifted in manifests.
- Retire --sgx and --sgx-key options, because:
    1) turning on/off SGX is not a global thing, user should be able to
       run any image in simulation mode at any time;
    2) gramine-direct was never working for flask anyway;
    3) in different gramine-sgx-sign plugins there may not be a --key at
       all.
- Replace --sgx-key with {% block sign_args %} in template and
  sgx.sign_args config in scag.toml.
- Update documentation.

Co-authored-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>
Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>
Signed-off-by: Wojtek Porczyk <woju@invisiblethingslab.com>

woju committed:
Scaffolding for Gramine 1.0~alpha1

Signed-off-by: Wojtek Porczyk <woju@invisiblethingslab.com>

woju committed:
README.rst: Describe instalation of unstable Gramine

Signed-off-by: Wojtek Porczyk <woju@invisiblethingslab.com>

woju committed:
Documentation/index.rst: include README.rst from repo root

Signed-off-by: Wojtek Porczyk <woju@invisiblethingslab.com>

woju committed:
debian/control: Recommend gramine-ratls-* packages

Signed-off-by: Wojtek Porczyk <woju@invisiblethingslab.com>

woju committed:
Add scag-client

Signed-off-by: Wojtek Porczyk <woju@invisiblethingslab.com>

woju committed:
Render scag-client.toml

Signed-off-by: Wojtek Porczyk <woju@invisiblethingslab.com>

woju committed:
tests/flask: use scag-client for an end-to-end test

Signed-off-by: Wojtek Porczyk <woju@invisiblethingslab.com>

woju committed:
templates: make debug and attestation configurable from scag.toml

Signed-off-by: Wojtek Porczyk <woju@invisiblethingslab.com>

woju committed:
Documentation: Describe signing with prod keys

Signed-off-by: Wojtek Porczyk <woju@invisiblethingslab.com>

woju committed:
zuul/build-deb: Don't attempt to sign

Signing will fail, CI has no keys for signing uploads.

Signed-off-by: Wojtek Porczyk <woju@invisiblethingslab.com>

woju committed:
debian: Fix lintian problem with unknown unstable- distros

There's no way to override tags in .changes, so --suppress-tags is the
only option. https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=575400

Signed-off-by: Wojtek Porczyk <woju@invisiblethingslab.com>

gramine-scaffolding commented:
Starting check jobs.

gramine-scaffolding commented:
Build succeeded.
https://zuul.gramineproject.io/t/scaffolding/buildset/bf95076dc40a4c9b9aaf63ba9d42de53

:heavy_check_mark: [bandit ](https://zuul.gramineproject.io/t/scaffolding/build/5c24e5f1f58841eeb9b98ff599505304) SUCCESS in 38s
:heavy_check_mark: [pylint ](https://zuul.gramineproject.io/t/scaffolding/build/af4d9bab078c462a8e152757ff5cc03e) SUCCESS in 47s
:heavy_check_mark: [pytest ](https://zuul.gramineproject.io/t/scaffolding/build/cbcdd7a91daf4f4c83831f51e9d06017) SUCCESS in 38s
:heavy_check_mark: [build-deb ](https://zuul.gramineproject.io/t/scaffolding/build/f88e29d8caef46e09b41526ce61a49b1) SUCCESS in 1m 03s
:heavy_check_mark: [build-docs ](https://zuul.gramineproject.io/t/scaffolding/build/00fd60151ca64f1a95f924e2b634353e) SUCCESS in 37s
:heavy_check_mark: [test-python ](https://zuul.gramineproject.io/t/scaffolding/build/067dc6260bb341278ad7361fb6abd421) SUCCESS in 2m 32s
:heavy_check_mark: [test-flask ](https://zuul.gramineproject.io/t/scaffolding/build/653da251041e44eb971f3b40ca463189) SUCCESS in 2m 53s
:heavy_check_mark: [test-nodejs ](https://zuul.gramineproject.io/t/scaffolding/build/2f605a3602d34d9eb07a0444cf1d7792) SUCCESS in 2m 55s

oshogbo committed:
Add passthrough environments

Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>

oshogbo committed:
Fix quickstart

The build step requires context.

Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>

oshogbo committed:
Fix typo in manpage

Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>

oshogbo committed:
Add github templates

Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>

oshogbo committed:
Add Java Jar framework

Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>

oshogbo committed:
Add Java with gradle builder

Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>

gramine-scaffolding commented:
Starting check jobs.

gramine-scaffolding commented:
Build succeeded.
https://zuul.gramineproject.io/t/scaffolding/buildset/f169e140c84a4b74b982a4926f858393

:heavy_check_mark: [bandit ](https://zuul.gramineproject.io/t/scaffolding/build/949f37a4645c4b36b2f3b3ee05f308d3) SUCCESS in 39s
:heavy_check_mark: [pylint ](https://zuul.gramineproject.io/t/scaffolding/build/201cd03da6074363a15c304d04e945e7) SUCCESS in 47s
:heavy_check_mark: [pytest ](https://zuul.gramineproject.io/t/scaffolding/build/45411f89e044423097ba2e9a432de485) SUCCESS in 38s
:heavy_check_mark: [build-deb ](https://zuul.gramineproject.io/t/scaffolding/build/2b46b34175184e45b7689ae5c0a50293) SUCCESS in 1m 02s
:heavy_check_mark: [build-docs ](https://zuul.gramineproject.io/t/scaffolding/build/94faf7e4fe38473ca2a2af673ae4fd3d) SUCCESS in 37s
:heavy_check_mark: [test-python ](https://zuul.gramineproject.io/t/scaffolding/build/ae64a624c8c1447d9ae787aa46390629) SUCCESS in 2m 57s
:heavy_check_mark: [test-flask ](https://zuul.gramineproject.io/t/scaffolding/build/8a6e8ccbe50b4990a20ce0679b5b4214) SUCCESS in 3m 10s
:heavy_check_mark: [test-nodejs ](https://zuul.gramineproject.io/t/scaffolding/build/ccde0cef6b4445c198894c0aa37ed06a) SUCCESS in 4m 46s

gramine-scaffolding commented:
Starting check jobs.

gramine-scaffolding commented:
Build succeeded.
https://zuul.gramineproject.io/t/scaffolding/buildset/7e9a253bbda74870804ad1adbc004626

:heavy_check_mark: [bandit ](https://zuul.gramineproject.io/t/scaffolding/build/91560543b38a4aeeb858d1d4f2655b6a) SUCCESS in 37s
:heavy_check_mark: [pylint ](https://zuul.gramineproject.io/t/scaffolding/build/dc74411343c842b5af313c86f9f306af) SUCCESS in 48s
:heavy_check_mark: [pytest ](https://zuul.gramineproject.io/t/scaffolding/build/5e723a3251464eb7a60ff1a1d200362a) SUCCESS in 38s
:heavy_check_mark: [build-deb ](https://zuul.gramineproject.io/t/scaffolding/build/c6fa883a40ac4870af3bd1f46dae50bb) SUCCESS in 1m 02s
:heavy_check_mark: [build-docs ](https://zuul.gramineproject.io/t/scaffolding/build/ce439d99b404466998a9f872b3838d3e) SUCCESS in 39s
:heavy_check_mark: [test-python ](https://zuul.gramineproject.io/t/scaffolding/build/1db87f3883c248d997846ec0ae672313) SUCCESS in 2m 38s
:heavy_check_mark: [test-flask ](https://zuul.gramineproject.io/t/scaffolding/build/fe0691b509ae445ca0398f2adea7e36b) SUCCESS in 2m 55s
:heavy_check_mark: [test-nodejs ](https://zuul.gramineproject.io/t/scaffolding/build/454bff6f05bf40ccaef688a8f80b7d7b) SUCCESS in 3m 04s

gramine-scaffolding commented:
Starting check jobs.

gramine-scaffolding commented:
Build succeeded.
https://zuul.gramineproject.io/t/scaffolding/buildset/26c29edadfa14549ac8b5374fb923068

:heavy_check_mark: [bandit ](https://zuul.gramineproject.io/t/scaffolding/build/059d37cf3bf741eebf00bc524d42b3e8) SUCCESS in 37s
:heavy_check_mark: [pylint ](https://zuul.gramineproject.io/t/scaffolding/build/9d6dd37c93334f6d8476a8c3d553b2d7) SUCCESS in 47s
:heavy_check_mark: [pytest ](https://zuul.gramineproject.io/t/scaffolding/build/69b2ae4a024e4ea0b99d6226efc53671) SUCCESS in 38s
:heavy_check_mark: [build-deb ](https://zuul.gramineproject.io/t/scaffolding/build/01e1787e0984483eb79c8f9427285a9e) SUCCESS in 1m 03s
:heavy_check_mark: [build-docs ](https://zuul.gramineproject.io/t/scaffolding/build/c31293433302423891c92d89405eb520) SUCCESS in 38s
:heavy_check_mark: [test-python ](https://zuul.gramineproject.io/t/scaffolding/build/fd3b839cd53c4cb1aa9da04736078420) SUCCESS in 2m 43s
:heavy_check_mark: [test-flask ](https://zuul.gramineproject.io/t/scaffolding/build/9eb37df4df45463a8fa8fb3ffb51040e) SUCCESS in 2m 58s
:heavy_check_mark: [test-nodejs ](https://zuul.gramineproject.io/t/scaffolding/build/9de89858980f41fa818abd999efe641d) SUCCESS in 2m 58s
:heavy_check_mark: [test-java-jar ](https://zuul.gramineproject.io/t/scaffolding/build/117768fb944642b1b62230cdb917cc15) SUCCESS in 4m 36s
:heavy_check_mark: [test-java-gradle ](https://zuul.gramineproject.io/t/scaffolding/build/70a55ac3e4394627aaba5d1ccf7783c4) SUCCESS in 6m 35s

gramine-scaffolding commented:
Starting check jobs.

gramine-scaffolding commented:
Build succeeded.
https://zuul.gramineproject.io/t/scaffolding/buildset/17cea6356a3740d1a76137978846e0fa

:heavy_check_mark: [bandit ](https://zuul.gramineproject.io/t/scaffolding/build/48e3c7263ec8439dadd1728be42c6720) SUCCESS in 37s
:heavy_check_mark: [pylint ](https://zuul.gramineproject.io/t/scaffolding/build/c396aa69b5a04840bb4d37ea74862bce) SUCCESS in 47s
:heavy_check_mark: [pytest ](https://zuul.gramineproject.io/t/scaffolding/build/2b9b263792624ba5b06f70ab2cfe360f) SUCCESS in 39s
:heavy_check_mark: [build-deb ](https://zuul.gramineproject.io/t/scaffolding/build/8c34e9b2491549c3afda6d09bc8490d6) SUCCESS in 1m 04s
:heavy_check_mark: [build-docs ](https://zuul.gramineproject.io/t/scaffolding/build/252638576c7b44bd8a48c0323af3726e) SUCCESS in 38s
:heavy_check_mark: [test-python ](https://zuul.gramineproject.io/t/scaffolding/build/8ca0d02531ab4dc1a0ad829159dee588) SUCCESS in 2m 46s
:heavy_check_mark: [test-flask ](https://zuul.gramineproject.io/t/scaffolding/build/beae6b4322544f9f8abf4f8717d40367) SUCCESS in 2m 51s
:heavy_check_mark: [test-nodejs ](https://zuul.gramineproject.io/t/scaffolding/build/8c18e2ed1fca47018f9bd9fcc3fccbb6) SUCCESS in 3m 05s
:heavy_check_mark: [test-java-jar ](https://zuul.gramineproject.io/t/scaffolding/build/245878f4fa18494ea962734da0824e87) SUCCESS in 4m 19s
:heavy_check_mark: [test-java-gradle ](https://zuul.gramineproject.io/t/scaffolding/build/9eddaa46013346d08bb570fe0cb333fc) SUCCESS in 6m 54s

gramine-scaffolding commented:
Starting check jobs.

gramine-scaffolding commented:
Build succeeded.
https://zuul.gramineproject.io/t/scaffolding/buildset/961e6c4816b04ae7a1e19dc723b24cbd

:heavy_check_mark: [bandit ](https://zuul.gramineproject.io/t/scaffolding/build/eda97dacc6914078a307637098b405ad) SUCCESS in 39s
:heavy_check_mark: [pylint ](https://zuul.gramineproject.io/t/scaffolding/build/fdf1cdf128474f4aa113ce577e34f7ad) SUCCESS in 49s
:heavy_check_mark: [pytest ](https://zuul.gramineproject.io/t/scaffolding/build/447c4db17c154345bfaec2f7e109bb5d) SUCCESS in 39s
:heavy_check_mark: [build-deb ](https://zuul.gramineproject.io/t/scaffolding/build/5a53a68ad031436ca6f036b35faf3048) SUCCESS in 1m 03s
:heavy_check_mark: [build-docs ](https://zuul.gramineproject.io/t/scaffolding/build/754296e92adf4e07891f9bb5265b2bc5) SUCCESS in 37s
:heavy_check_mark: [test-python ](https://zuul.gramineproject.io/t/scaffolding/build/dc0677aff95047848efa67ef02747997) SUCCESS in 2m 43s
:heavy_check_mark: [test-flask ](https://zuul.gramineproject.io/t/scaffolding/build/bb5dd98b6b494f14aff83ed8a38264fb) SUCCESS in 3m 01s
:heavy_check_mark: [test-nodejs ](https://zuul.gramineproject.io/t/scaffolding/build/c62228278fdd4f7a8d6dbd07438e3e51) SUCCESS in 3m 06s
:heavy_check_mark: [test-java-jar ](https://zuul.gramineproject.io/t/scaffolding/build/23807ac9165349bfa073156e3833c7e1) SUCCESS in 6m 05s
:heavy_check_mark: [test-java-gradle ](https://zuul.gramineproject.io/t/scaffolding/build/69125ebbf173427c9d4b1366df7f4df9) SUCCESS in 7m 43s

woju committed:
Fix mmdebstrap dependency

Signed-off-by: Wojtek Porczyk <woju@invisiblethingslab.com>

woju committed:
keys: install debian-archive-keyring for ubuntu <= 22.04

Ubuntu up to and including jammy (22.04LTS) does not ship keys for
bookworm.

Signed-off-by: Wojtek Porczyk <woju@invisiblethingslab.com>

woju committed:
builder: sign outside of mmdebstrap

Signing in mmdebstrap is essentially impossible, because:
- in --mode=sudo it attempts key from /root/.config/gramine, which is
  not what we want;
- in --mode=unshare $HOME/.config/gramine/enclave-key.pem is EACCES,
  because unshare does it's job before hook gets to run;
- in --mode=fakechroot some exotic buildchains may not work (i.e.
  dotnet) which sometimes does not link against libc (fakechroot works
  by LD_PRELOAD-ing some hooks that mangle paths in file-related
  syscalls before offloading to "real" syscalls).

Signed-off-by: Wojtek Porczyk <woju@invisiblethingslab.com>

gramine-scaffolding commented:
Starting check jobs.

gramine-scaffolding commented:
Build succeeded.
https://zuul.gramineproject.io/t/scaffolding/buildset/efd4937972d7462e847160b8a06bc91c

:heavy_check_mark: [bandit ](https://zuul.gramineproject.io/t/scaffolding/build/f59918ec96094a68ad3ead253fdbc180) SUCCESS in 39s
:heavy_check_mark: [pylint ](https://zuul.gramineproject.io/t/scaffolding/build/2ff048afb0d946b8804918734e4ba431) SUCCESS in 49s
:heavy_check_mark: [pytest ](https://zuul.gramineproject.io/t/scaffolding/build/b75e210c27654dd6a7f55d45bb420466) SUCCESS in 40s
:heavy_check_mark: [build-deb ](https://zuul.gramineproject.io/t/scaffolding/build/5a997fe9667f4d3da07970bc6873f444) SUCCESS in 1m 05s
:heavy_check_mark: [build-docs ](https://zuul.gramineproject.io/t/scaffolding/build/3ed7ddc6f76a43dc95c863308c58304c) SUCCESS in 37s
:heavy_check_mark: [test-python ](https://zuul.gramineproject.io/t/scaffolding/build/def7e6d2bf1a4e4eaeee40b347fe9a4f) SUCCESS in 2m 47s
:heavy_check_mark: [test-flask ](https://zuul.gramineproject.io/t/scaffolding/build/b4bc34cebab64276973aa30fabe6ea88) SUCCESS in 3m 04s
:heavy_check_mark: [test-nodejs ](https://zuul.gramineproject.io/t/scaffolding/build/72036c0f057f4670814e6c3af9d1f852) SUCCESS in 3m 13s
:heavy_check_mark: [test-java-jar ](https://zuul.gramineproject.io/t/scaffolding/build/309452e828374d2db8722fb2e7845df5) SUCCESS in 5m 33s
:heavy_check_mark: [test-java-gradle ](https://zuul.gramineproject.io/t/scaffolding/build/faa6143afc39407aae65886342c9ff36) SUCCESS in 10m 03s

gramine-scaffolding commented:
Starting check jobs.

gramine-scaffolding commented:
Build canceled.

gramine-scaffolding commented:
Starting check jobs.

gramine-scaffolding commented:
Build succeeded.
https://zuul.gramineproject.io/t/scaffolding/buildset/a246e1864ae8495c973b6be409c29bb1

:heavy_check_mark: [bandit ](https://zuul.gramineproject.io/t/scaffolding/build/74cb84237c67417fa8d3254a826a348b) SUCCESS in 38s
:heavy_check_mark: [pylint ](https://zuul.gramineproject.io/t/scaffolding/build/05c18a5ab80b46e3a86ca38e9cba86ed) SUCCESS in 48s
:heavy_check_mark: [pytest ](https://zuul.gramineproject.io/t/scaffolding/build/8a110b86f98e4ac4a7a2cbfa0d100b35) SUCCESS in 39s
:heavy_check_mark: [build-deb ](https://zuul.gramineproject.io/t/scaffolding/build/3f6ca44c803e4aadbc4d4ed729317592) SUCCESS in 1m 04s
:heavy_check_mark: [build-docs ](https://zuul.gramineproject.io/t/scaffolding/build/a1c45e13fed749d48ad75b3205acf027) SUCCESS in 37s
:heavy_check_mark: [test-python ](https://zuul.gramineproject.io/t/scaffolding/build/577812fa0dd143ea8940a4ecdbe37bf1) SUCCESS in 2m 58s
:heavy_check_mark: [test-flask ](https://zuul.gramineproject.io/t/scaffolding/build/32e0932caa5f4da4a92be7d3807ed4a5) SUCCESS in 3m 05s
:heavy_check_mark: [test-nodejs ](https://zuul.gramineproject.io/t/scaffolding/build/b350ad6563ff483d8eed71334ec0407d) SUCCESS in 3m 19s
:heavy_check_mark: [test-java-jar ](https://zuul.gramineproject.io/t/scaffolding/build/7182ecf615394cb2bfb6d02dea3f68ea) SUCCESS in 7m 27s
:heavy_check_mark: [test-java-gradle ](https://zuul.gramineproject.io/t/scaffolding/build/98cfaabefe0741c49d16877a7aa2680c) SUCCESS in 10m 37s

gramine-scaffolding commented:
Starting check jobs.

gramine-scaffolding commented:
Build canceled.

gramine-scaffolding commented:
Starting check jobs.

gramine-scaffolding commented:
Build canceled.

gramine-scaffolding commented:
Starting check jobs.

gramine-scaffolding commented:
Build succeeded.
https://zuul.gramineproject.io/t/scaffolding/buildset/3713981303d5482b8c214c392d1c8775

:heavy_check_mark: [bandit ](https://zuul.gramineproject.io/t/scaffolding/build/62f71a2dfba04dd7b3896dcc3462cbc6) SUCCESS in 39s
:heavy_check_mark: [pylint ](https://zuul.gramineproject.io/t/scaffolding/build/c980a697d19f4cdeab641679615afdc7) SUCCESS in 51s
:heavy_check_mark: [pytest ](https://zuul.gramineproject.io/t/scaffolding/build/01a4593857d64c0c96f1d2f12c402f50) SUCCESS in 39s
:heavy_check_mark: [build-deb ](https://zuul.gramineproject.io/t/scaffolding/build/ffa647bbc0f04789b431c58ec20a6f2f) SUCCESS in 1m 06s
:heavy_check_mark: [build-docs ](https://zuul.gramineproject.io/t/scaffolding/build/83d5edf124834c07b126830fe06357e0) SUCCESS in 38s
:heavy_check_mark: [test-python ](https://zuul.gramineproject.io/t/scaffolding/build/1c941b08d428416d867e6737122f4fe5) SUCCESS in 2m 26s
:heavy_check_mark: [test-flask ](https://zuul.gramineproject.io/t/scaffolding/build/733799d92c61443eb30fda0f79713e12) SUCCESS in 3m 09s
:heavy_check_mark: [test-nodejs ](https://zuul.gramineproject.io/t/scaffolding/build/3426ca3ca69f4e88acb75ab78ab217d1) SUCCESS in 3m 25s
:heavy_check_mark: [test-expressjs ](https://zuul.gramineproject.io/t/scaffolding/build/0c11e502b0284813a46bf15d97d960b5) SUCCESS in 5m 40s
:heavy_check_mark: [test-koajs ](https://zuul.gramineproject.io/t/scaffolding/build/1310f7f5ffaa4dee9288e593227337f8) SUCCESS in 5m 26s
:heavy_check_mark: [test-java-jar ](https://zuul.gramineproject.io/t/scaffolding/build/3c387bbcb47348af94c34927dc90693e) SUCCESS in 4m 01s
:heavy_check_mark: [test-java-gradle ](https://zuul.gramineproject.io/t/scaffolding/build/1d7a30df665847d79b36f60787e30667) SUCCESS in 6m 28s
:heavy_check_mark: [test-dotnet ](https://zuul.gramineproject.io/t/scaffolding/build/2cfa4f9ecd48472b815ea8652faf88ed) SUCCESS in 6m 36s

oshogbo committed:
Fix printing help

Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>

woju committed:
builder: Allow custom templates directories in config and plugins

Signed-off-by: Wojtek Porczyk <woju@invisiblethingslab.com>

woju committed:
builder: Allow customisations through Dockefile

Signed-off-by: Wojtek Porczyk <woju@invisiblethingslab.com>

woju committed:
Use stable gramine (1.6), which is already released

Signed-off-by: Wojtek Porczyk <woju@invisiblethingslab.com>

gramine-scaffolding commented:
Starting check jobs.

gramine-scaffolding commented:
Build succeeded.
https://zuul.gramineproject.io/t/scaffolding/buildset/0bbaaa0be9ec4deba3d8dfe31e494bfe

:heavy_check_mark: [bandit ](https://zuul.gramineproject.io/t/scaffolding/build/5f40962ef7dc4a978590954ad98e1af9) SUCCESS in 38s
:heavy_check_mark: [pylint ](https://zuul.gramineproject.io/t/scaffolding/build/e55b22f280974b0083c31507e8b47228) SUCCESS in 49s
:heavy_check_mark: [pytest ](https://zuul.gramineproject.io/t/scaffolding/build/e851444cec3649c3a900d647ea55cbad) SUCCESS in 38s
:heavy_check_mark: [build-deb ](https://zuul.gramineproject.io/t/scaffolding/build/78a99fccaa3948cd827289bbc63de0d2) SUCCESS in 1m 05s
:heavy_check_mark: [build-docs ](https://zuul.gramineproject.io/t/scaffolding/build/be3e3f8a26e44f41aad62d36a1724ca0) SUCCESS in 38s
:heavy_check_mark: [test-python ](https://zuul.gramineproject.io/t/scaffolding/build/d5a9ce6b10e24ba28205ca5bd15958fb) SUCCESS in 2m 43s
:heavy_check_mark: [test-flask ](https://zuul.gramineproject.io/t/scaffolding/build/eb61993f03a74cf398de74183f9561b6) SUCCESS in 2m 52s
:heavy_check_mark: [test-nodejs ](https://zuul.gramineproject.io/t/scaffolding/build/1c4b614b258d44218ef176583d49a2c4) SUCCESS in 3m 29s
:heavy_check_mark: [test-expressjs ](https://zuul.gramineproject.io/t/scaffolding/build/b8d42a7a7d7e454789b262fc74138d07) SUCCESS in 5m 16s
:heavy_check_mark: [test-koajs ](https://zuul.gramineproject.io/t/scaffolding/build/6fd0529c46d944009b398b673753b450) SUCCESS in 5m 06s
:heavy_check_mark: [test-java-jar ](https://zuul.gramineproject.io/t/scaffolding/build/281ad933c00b4c468a38e6274b14935d) SUCCESS in 4m 19s
:heavy_check_mark: [test-java-gradle ](https://zuul.gramineproject.io/t/scaffolding/build/8e184a5638b54398b1ca61e1b73a71e1) SUCCESS in 7m 47s
:heavy_check_mark: [test-dotnet ](https://zuul.gramineproject.io/t/scaffolding/build/ef4f7ce4803048e28622078bda2021ff) SUCCESS in 6m 46s

gramine-scaffolding commented:
Starting check jobs.

gramine-scaffolding commented:
Build failed.
https://zuul.gramineproject.io/t/scaffolding/buildset/83c220bcf6954683949b9062fe51802e

:heavy_check_mark: [bandit ](https://zuul.gramineproject.io/t/scaffolding/build/be9fee12f54846e2a003ab413f1934b7) SUCCESS in 38s
:heavy_check_mark: [pylint ](https://zuul.gramineproject.io/t/scaffolding/build/1e96c3d3e1ee44098285b09663c8cd69) SUCCESS in 49s
:heavy_check_mark: [pytest ](https://zuul.gramineproject.io/t/scaffolding/build/a789bdfd7190478e95e6c4cde8a260d7) SUCCESS in 39s
:heavy_check_mark: [build-deb ](https://zuul.gramineproject.io/t/scaffolding/build/ac025c343c4341d9a4efbc3fbd0ea191) SUCCESS in 1m 05s
:heavy_check_mark: [build-docs ](https://zuul.gramineproject.io/t/scaffolding/build/ab081d0246fe449aa90e3091e82c5d06) SUCCESS in 39s
:heavy_check_mark: [test-python ](https://zuul.gramineproject.io/t/scaffolding/build/32c8b7deb9d44030a59341e047bf8b0f) SUCCESS in 2m 50s
:heavy_check_mark: [test-flask ](https://zuul.gramineproject.io/t/scaffolding/build/4c9b5d376ed84b4e841c603c11b36428) SUCCESS in 2m 55s
:heavy_check_mark: [test-nodejs ](https://zuul.gramineproject.io/t/scaffolding/build/d99bd463b11844f99abfd32194f51cea) SUCCESS in 3m 06s
:heavy_check_mark: [test-expressjs ](https://zuul.gramineproject.io/t/scaffolding/build/35ec892e29c6450084d601c6074ebc34) SUCCESS in 5m 28s
:heavy_check_mark: [test-koajs ](https://zuul.gramineproject.io/t/scaffolding/build/a90384ff300f46aeb687ce3793e89489) SUCCESS in 5m 25s
:heavy_check_mark: [test-java-jar ](https://zuul.gramineproject.io/t/scaffolding/build/a3232fc723ec4d8684949e3862123bf9) SUCCESS in 4m 20s
:heavy_check_mark: [test-java-gradle ](https://zuul.gramineproject.io/t/scaffolding/build/55c76c0bdbd949ba80f84d19889a471e) SUCCESS in 7m 48s
:x: [test-dotnet ](https://zuul.gramineproject.io/t/scaffolding/build/69a26edea156453581b1ebc017618fec) **FAILURE** in 3m 20s

woju committed:
Extend expect timeout for dotnet to 10 min

Signed-off-by: Wojtek Porczyk <woju@invisiblethingslab.com>

gramine-scaffolding commented:
Starting check jobs.

gramine-scaffolding commented:
Build succeeded.
https://zuul.gramineproject.io/t/scaffolding/buildset/5bcf83e68c754beaa2d3ea5f31ef1d37

:heavy_check_mark: [bandit ](https://zuul.gramineproject.io/t/scaffolding/build/2fd60e7baccf42dd9fa06fd1b85f13ff) SUCCESS in 39s
:heavy_check_mark: [pylint ](https://zuul.gramineproject.io/t/scaffolding/build/86d76d0d643d43b99c55cb5507f660f8) SUCCESS in 49s
:heavy_check_mark: [pytest ](https://zuul.gramineproject.io/t/scaffolding/build/a3cecf57f0574a389caa5e9610328576) SUCCESS in 38s
:heavy_check_mark: [build-deb ](https://zuul.gramineproject.io/t/scaffolding/build/8f874d7d78754314a5b24b26c056b58f) SUCCESS in 1m 05s
:heavy_check_mark: [build-docs ](https://zuul.gramineproject.io/t/scaffolding/build/f15ff66a42fd4d93951ed17b5660e430) SUCCESS in 37s
:heavy_check_mark: [test-python ](https://zuul.gramineproject.io/t/scaffolding/build/7e76fe24c49248d7b7fcfe9c3b6d1b24) SUCCESS in 2m 50s
:heavy_check_mark: [test-flask ](https://zuul.gramineproject.io/t/scaffolding/build/af077c77534d491fa50551c1fe1cb86d) SUCCESS in 4m 09s
:heavy_check_mark: [test-nodejs ](https://zuul.gramineproject.io/t/scaffolding/build/e9bf4813e6bc481d92df9a335afd6420) SUCCESS in 3m 22s
:heavy_check_mark: [test-expressjs ](https://zuul.gramineproject.io/t/scaffolding/build/48fa16caa1f440489372ecc88c1e0405) SUCCESS in 6m 17s
:heavy_check_mark: [test-koajs ](https://zuul.gramineproject.io/t/scaffolding/build/45dbbf7ae9a04bd0bb3d1ab93f398afb) SUCCESS in 7m 09s
:heavy_check_mark: [test-java-jar ](https://zuul.gramineproject.io/t/scaffolding/build/8aeb4abc378f489bbef598d91241a101) SUCCESS in 4m 30s
:heavy_check_mark: [test-java-gradle ](https://zuul.gramineproject.io/t/scaffolding/build/321daedc10514b1584f9e9beabc11268) SUCCESS in 8m 55s
:heavy_check_mark: [test-dotnet ](https://zuul.gramineproject.io/t/scaffolding/build/751eed252b034c478beb8a92bf237564) SUCCESS in 7m 45s

oshogbo committed:
Ensure tomli.load uses files opened in binary read mode

Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>

gramine-scaffolding commented:
Starting check jobs.

gramine-scaffolding commented:
Build succeeded.
https://zuul.gramineproject.io/t/scaffolding/buildset/d3ff823039224c1c99e4be121319455f

:heavy_check_mark: [bandit ](https://zuul.gramineproject.io/t/scaffolding/build/5489bbbaefea4ee9abd409b8002a63ec) SUCCESS in 38s
:heavy_check_mark: [pylint ](https://zuul.gramineproject.io/t/scaffolding/build/52e0753963ba4690b304152fdc1d8ffb) SUCCESS in 49s
:heavy_check_mark: [pytest ](https://zuul.gramineproject.io/t/scaffolding/build/045ba023797f46479e43ac63e152a7ac) SUCCESS in 39s
:heavy_check_mark: [build-deb ](https://zuul.gramineproject.io/t/scaffolding/build/060a167e9c1c49739fea16bef56b544b) SUCCESS in 1m 04s
:heavy_check_mark: [build-docs ](https://zuul.gramineproject.io/t/scaffolding/build/86ead1b645d649f880b9b87c7a934bc5) SUCCESS in 37s
:heavy_check_mark: [test-python ](https://zuul.gramineproject.io/t/scaffolding/build/e7bdf975b24d441e8badcb8ff21363da) SUCCESS in 2m 44s
:heavy_check_mark: [test-flask ](https://zuul.gramineproject.io/t/scaffolding/build/def26224677d48a9a55d5d7c6fe2097f) SUCCESS in 4m 18s
:heavy_check_mark: [test-nodejs ](https://zuul.gramineproject.io/t/scaffolding/build/0ce3d68d41e84b58b40151b84c89b117) SUCCESS in 2m 59s
:heavy_check_mark: [test-expressjs ](https://zuul.gramineproject.io/t/scaffolding/build/807188d58b7348129764ab1db2015aad) SUCCESS in 6m 30s
:heavy_check_mark: [test-koajs ](https://zuul.gramineproject.io/t/scaffolding/build/fb6c1dc123934380aa91b2e2fc52c0d2) SUCCESS in 6m 50s
:heavy_check_mark: [test-java-jar ](https://zuul.gramineproject.io/t/scaffolding/build/47994d9cde574a5fac9bdce20e90c7e3) SUCCESS in 6m 54s
:heavy_check_mark: [test-java-gradle ](https://zuul.gramineproject.io/t/scaffolding/build/e58fdb7529ba44e49572aacd804c362c) SUCCESS in 8m 34s
:heavy_check_mark: [test-dotnet ](https://zuul.gramineproject.io/t/scaffolding/build/460f07acb2ff4e5bb482a5fa70a787b3) SUCCESS in 8m 05s

gramine-scaffolding commented:
Starting check jobs.

gramine-scaffolding commented:
Build succeeded.
https://zuul.gramineproject.io/t/scaffolding/buildset/25b56ffe9bac4b4d8ae6385f95a4ed68

:heavy_check_mark: [bandit ](https://zuul.gramineproject.io/t/scaffolding/build/7b9c1e4330cf43d1b52524c9ce9fed23) SUCCESS in 38s
:heavy_check_mark: [pylint ](https://zuul.gramineproject.io/t/scaffolding/build/d82446486d4a4a8bbf46661abe7fdb2c) SUCCESS in 50s
:heavy_check_mark: [pytest ](https://zuul.gramineproject.io/t/scaffolding/build/28db928575e348e29f0cbd5ae8343427) SUCCESS in 38s
:heavy_check_mark: [build-deb ](https://zuul.gramineproject.io/t/scaffolding/build/c7dc1ca70a2c4bc3ab08f2aea459d648) SUCCESS in 1m 05s
:heavy_check_mark: [build-docs ](https://zuul.gramineproject.io/t/scaffolding/build/52839c7ab39748ed8a3cf7900cd7ada4) SUCCESS in 37s
:heavy_check_mark: [test-python ](https://zuul.gramineproject.io/t/scaffolding/build/399a59a4667f422e9d93250a245b20a7) SUCCESS in 3m 13s
:heavy_check_mark: [test-flask ](https://zuul.gramineproject.io/t/scaffolding/build/e3d93e5ac19242eba33f32e2e87345e3) SUCCESS in 5m 21s
:heavy_check_mark: [test-nodejs ](https://zuul.gramineproject.io/t/scaffolding/build/41e97b8f7f734044b04e6420a0e6ccb2) SUCCESS in 5m 40s
:heavy_check_mark: [test-expressjs ](https://zuul.gramineproject.io/t/scaffolding/build/df554c21256e40c8b7ddcc2d8fd96ada) SUCCESS in 7m 34s
:heavy_check_mark: [test-koajs ](https://zuul.gramineproject.io/t/scaffolding/build/694a993f03904838a50d1c9b350523ff) SUCCESS in 8m 20s
:heavy_check_mark: [test-java-jar ](https://zuul.gramineproject.io/t/scaffolding/build/348e0134adb448db888244c2c28a6db3) SUCCESS in 6m 44s
:heavy_check_mark: [test-java-gradle ](https://zuul.gramineproject.io/t/scaffolding/build/fa1f0220aa3946308236a54e49a8c4b0) SUCCESS in 16m 01s
:heavy_check_mark: [test-dotnet ](https://zuul.gramineproject.io/t/scaffolding/build/679241c0cb69414190e197a0611d0d11) SUCCESS in 11m 14s

woju committed:
Documentation: Add templates.rst

Signed-off-by: Wojtek Porczyk <woju@invisiblethingslab.com>

gramine-scaffolding commented:
Starting check jobs.

gramine-scaffolding commented:
Build failed.
https://zuul.gramineproject.io/t/scaffolding/buildset/eced043abdd2476f9bb3c5b20d30547f

:heavy_check_mark: [bandit ](https://zuul.gramineproject.io/t/scaffolding/build/215cd45e0f27473d8dd9445f6ba84667) SUCCESS in 37s
:heavy_check_mark: [pylint ](https://zuul.gramineproject.io/t/scaffolding/build/9b92cafcbc4d4b88bebc1d9ef00faca8) SUCCESS in 50s
:heavy_check_mark: [pytest ](https://zuul.gramineproject.io/t/scaffolding/build/e6a233e3c0cc46fa8bee15518b4ac4e2) SUCCESS in 38s
:heavy_check_mark: [build-deb ](https://zuul.gramineproject.io/t/scaffolding/build/21e41c07bf6841b5a535197ffc5012df) SUCCESS in 1m 06s
:heavy_check_mark: [build-docs ](https://zuul.gramineproject.io/t/scaffolding/build/ada5e597cc0c4d6f9b2abc8e0f322fe8) SUCCESS in 38s
:x: [test-python ](https://zuul.gramineproject.io/t/scaffolding/build/f6410bc412c04d2e8528d1ad9209fda3) **FAILURE** in 6m 22s
:heavy_check_mark: [test-flask ](https://zuul.gramineproject.io/t/scaffolding/build/b01396621ac1421e8fd1b131d19aa8f8) SUCCESS in 5m 21s
:heavy_check_mark: [test-nodejs ](https://zuul.gramineproject.io/t/scaffolding/build/e14af6cbc6664086a48fac10490fd520) SUCCESS in 3m 16s
:heavy_check_mark: [test-expressjs ](https://zuul.gramineproject.io/t/scaffolding/build/14f9af56ac324ee58c3a61112148da00) SUCCESS in 6m 07s
:heavy_check_mark: [test-koajs ](https://zuul.gramineproject.io/t/scaffolding/build/19856d5a8b0041b9ba1d05d011b7a077) SUCCESS in 5m 43s
:heavy_check_mark: [test-java-jar ](https://zuul.gramineproject.io/t/scaffolding/build/fb035f1494744b3d9e6e82eb9802405c) SUCCESS in 4m 41s
:heavy_check_mark: [test-java-gradle ](https://zuul.gramineproject.io/t/scaffolding/build/87e35a533b5b4c19a936ccffba47cd0c) SUCCESS in 11m 58s
:heavy_check_mark: [test-dotnet ](https://zuul.gramineproject.io/t/scaffolding/build/792f1ccc842b46619f0c155f6ef61eec) SUCCESS in 8m 08s

oshogbo committed:
run: add information about ports

Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>
Co-authored-by: Wojtek Porczyk <woju@invisiblethingslab.com>
Signed-off-by: Wojtek Porczyk <woju@invisiblethingslab.com>

oshogbo committed:
Use unix domain sockets instead of ports

Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>

woju committed:
client: Fix scag-client for MAA attestattion

--allow:
    When click has --yes/--no option, it's False by default, but we
    shouldn't pass it at all if not specified, hence we should default to
    None. The default will be set in client.py, in the function signature.
maa_provider_url:
    I don't know any other providers. Lack of this default meant user
    needed to provide this URL in scag-client.toml every time.
scag-client.toml:
    As a reminder how to set this option, if needed. Some hardware will
    require it, and it's up to programmer to verify the application
    him/herself.

Signed-off-by: Wojtek Porczyk <woju@invisiblethingslab.com>

woju committed:
tests: Extend dotnet timeout yet again

Signed-off-by: Wojtek Porczyk <woju@invisiblethingslab.com>

gramine-scaffolding commented:
Starting check jobs.

gramine-scaffolding commented:
Build succeeded.
https://zuul.gramineproject.io/t/scaffolding/buildset/68617f4fa9004476877fe1800a05950c

:heavy_check_mark: [bandit ](https://zuul.gramineproject.io/t/scaffolding/build/12dd139fed3e437c8f457a6697149faf) SUCCESS in 39s
:heavy_check_mark: [pylint ](https://zuul.gramineproject.io/t/scaffolding/build/7b8f6f884aba4610bba741d29c4004b7) SUCCESS in 50s
:heavy_check_mark: [pytest ](https://zuul.gramineproject.io/t/scaffolding/build/c58ac68ce2064b3bb71b038b811fcb5f) SUCCESS in 39s
:heavy_check_mark: [build-deb ](https://zuul.gramineproject.io/t/scaffolding/build/2fff3cd2488f4dc0bf3e3b6f11c5f5f0) SUCCESS in 1m 06s
:heavy_check_mark: [build-docs ](https://zuul.gramineproject.io/t/scaffolding/build/2773c63fa11a40c79e2912c3d7ba6ee1) SUCCESS in 37s
:heavy_check_mark: [test-python ](https://zuul.gramineproject.io/t/scaffolding/build/054b9125c484413fb91c284bba1483ea) SUCCESS in 3m 32s
:heavy_check_mark: [test-flask ](https://zuul.gramineproject.io/t/scaffolding/build/a7809b69067d4cf6bad27658ea50bb06) SUCCESS in 4m 03s
:heavy_check_mark: [test-nodejs ](https://zuul.gramineproject.io/t/scaffolding/build/0ed6d956b6f54b59a86e214906093bd3) SUCCESS in 5m 11s
:heavy_check_mark: [test-expressjs ](https://zuul.gramineproject.io/t/scaffolding/build/21424aa11d88413f975a963a39d8c20e) SUCCESS in 7m 07s
:heavy_check_mark: [test-koajs ](https://zuul.gramineproject.io/t/scaffolding/build/05256e1c55a14bafb7d8c5e190461395) SUCCESS in 7m 39s
:heavy_check_mark: [test-java-jar ](https://zuul.gramineproject.io/t/scaffolding/build/f47af975b64943e5a3946369da0e0e48) SUCCESS in 8m 15s
:heavy_check_mark: [test-java-gradle ](https://zuul.gramineproject.io/t/scaffolding/build/1f95c890516042fc912425d1e9d6e870) SUCCESS in 13m 18s
:heavy_check_mark: [test-dotnet ](https://zuul.gramineproject.io/t/scaffolding/build/4bb70e60fef942d59d56a7fa717eedb1) SUCCESS in 11m 52s

gramine-scaffolding commented:
Starting check jobs.

gramine-scaffolding commented:
Build succeeded.
https://zuul.gramineproject.io/t/scaffolding/buildset/087b355cddfe4cda9cd2b762d003fa21

:heavy_check_mark: [bandit ](https://zuul.gramineproject.io/t/scaffolding/build/632b24ef4ff249068c4f390cb44a29df) SUCCESS in 38s
:heavy_check_mark: [pylint ](https://zuul.gramineproject.io/t/scaffolding/build/a0b5e81105dc4b71981d2e2d6c361abb) SUCCESS in 49s
:heavy_check_mark: [pytest ](https://zuul.gramineproject.io/t/scaffolding/build/26fbb15ec3124afa8a5449b6c838ea9f) SUCCESS in 38s
:heavy_check_mark: [build-deb ](https://zuul.gramineproject.io/t/scaffolding/build/1073e00b9e704ca6861c2bd151cfcb75) SUCCESS in 1m 07s
:heavy_check_mark: [build-docs ](https://zuul.gramineproject.io/t/scaffolding/build/b4037397634f4f14849472a99f10530d) SUCCESS in 38s
:heavy_check_mark: [test-python ](https://zuul.gramineproject.io/t/scaffolding/build/fde581874d634bfcb32a5c36bd6fbf1d) SUCCESS in 3m 12s
:heavy_check_mark: [test-flask ](https://zuul.gramineproject.io/t/scaffolding/build/74b96c1a7434462e995e04c88e416b91) SUCCESS in 3m 00s
:heavy_check_mark: [test-nodejs ](https://zuul.gramineproject.io/t/scaffolding/build/c06c68d9e38848388c4d7436313cbd1a) SUCCESS in 3m 21s
:heavy_check_mark: [test-expressjs ](https://zuul.gramineproject.io/t/scaffolding/build/289d791977dd42df8e168cbf5bfada0f) SUCCESS in 5m 36s
:heavy_check_mark: [test-koajs ](https://zuul.gramineproject.io/t/scaffolding/build/b4e6880a1d2d4686900ec2a96934cc3d) SUCCESS in 6m 34s
:heavy_check_mark: [test-java-jar ](https://zuul.gramineproject.io/t/scaffolding/build/60d806161e014126bff4afe74b78563d) SUCCESS in 6m 24s
:heavy_check_mark: [test-java-gradle ](https://zuul.gramineproject.io/t/scaffolding/build/aa1e2fde21bb49bc9781c9a878cb909d) SUCCESS in 10m 17s
:heavy_check_mark: [test-dotnet ](https://zuul.gramineproject.io/t/scaffolding/build/340fdeb33d1e420695abd1b95b621bf3) SUCCESS in 8m 54s

gramine-scaffolding commented:
Starting check jobs.

woju committed:
Gramine-scaffolding 0.1

Signed-off-by: Wojtek Porczyk <woju@invisiblethingslab.com>

gramine-scaffolding commented:
Build failed.
https://zuul.gramineproject.io/t/scaffolding/buildset/b293e7de81994fbe90dc5412ad8df8cc

:heavy_check_mark: [bandit ](https://zuul.gramineproject.io/t/scaffolding/build/2ee59c793cdd48dc96082224bf6c7733) SUCCESS in 39s
:heavy_check_mark: [pylint ](https://zuul.gramineproject.io/t/scaffolding/build/fdda67e677224aacad10c8a7afb22414) SUCCESS in 51s
:heavy_check_mark: [pytest ](https://zuul.gramineproject.io/t/scaffolding/build/7978b886409f412180068f3a80fd747f) SUCCESS in 39s
:heavy_check_mark: [build-deb ](https://zuul.gramineproject.io/t/scaffolding/build/09f6d1f4afe74fb2a3519000a4b96802) SUCCESS in 1m 05s
:heavy_check_mark: [build-docs ](https://zuul.gramineproject.io/t/scaffolding/build/e63eccb832f44b8092a8692c2246e3c3) SUCCESS in 39s
:heavy_check_mark: [test-python ](https://zuul.gramineproject.io/t/scaffolding/build/38642d7292f7429c90a0a68d8f5a9714) SUCCESS in 4m 39s
:heavy_check_mark: [test-flask ](https://zuul.gramineproject.io/t/scaffolding/build/57e17fc7113f418686057c6910fba975) SUCCESS in 3m 57s
:heavy_check_mark: [test-nodejs ](https://zuul.gramineproject.io/t/scaffolding/build/4029399cc18f4ca4b88f51915335c610) SUCCESS in 4m 06s
:heavy_check_mark: [test-expressjs ](https://zuul.gramineproject.io/t/scaffolding/build/5a7b0000b822403786b35b538565f68b) SUCCESS in 8m 35s
:heavy_check_mark: [test-koajs ](https://zuul.gramineproject.io/t/scaffolding/build/d9595a21e3db4128b3fdf1b70663c295) SUCCESS in 9m 00s
:heavy_check_mark: [test-java-jar ](https://zuul.gramineproject.io/t/scaffolding/build/6a4acff69ae74d9281142392c1ebc0cc) SUCCESS in 6m 11s
:x: [test-java-gradle ](https://zuul.gramineproject.io/t/scaffolding/build/7e0fb82be7064f7abc0e637e95b61ee6) **FAILURE** in 10m 27s
:x: [test-dotnet ](https://zuul.gramineproject.io/t/scaffolding/build/3cd2abaf560d4399a697647dbaea2591) **FAILURE** in 8m 38s

gramine-scaffolding commented:
Starting check jobs.

gramine-scaffolding commented:
Build succeeded.
https://zuul.gramineproject.io/t/scaffolding/buildset/f6d6d1319f91480e9a0b6f9f78d7f08e

:heavy_check_mark: [bandit ](https://zuul.gramineproject.io/t/scaffolding/build/772e959f2b8a42b1849ce2276af4d85b) SUCCESS in 40s
:heavy_check_mark: [pylint ](https://zuul.gramineproject.io/t/scaffolding/build/8656c6d7b8a14b3fa9b5f751b987d255) SUCCESS in 51s
:heavy_check_mark: [pytest ](https://zuul.gramineproject.io/t/scaffolding/build/21886a264fbb4381ad67bbfdb5615e07) SUCCESS in 38s
:heavy_check_mark: [build-deb ](https://zuul.gramineproject.io/t/scaffolding/build/f4e60bd781d74822a0d662926d793755) SUCCESS in 1m 06s
:heavy_check_mark: [build-docs ](https://zuul.gramineproject.io/t/scaffolding/build/ade69bd5db4f46adb6fd2f99949d33b6) SUCCESS in 38s
:heavy_check_mark: [test-python ](https://zuul.gramineproject.io/t/scaffolding/build/a71b2ab36d384f5080d3f8825f382c1f) SUCCESS in 5m 09s
:heavy_check_mark: [test-flask ](https://zuul.gramineproject.io/t/scaffolding/build/f627d2f040854db892f6f15feabe79c3) SUCCESS in 4m 24s
:heavy_check_mark: [test-nodejs ](https://zuul.gramineproject.io/t/scaffolding/build/447d1b0566b9450cb2b81678e1f25255) SUCCESS in 5m 56s
:heavy_check_mark: [test-expressjs ](https://zuul.gramineproject.io/t/scaffolding/build/1543743a62ab48aeacc96c4e5d760cdb) SUCCESS in 7m 52s
:heavy_check_mark: [test-koajs ](https://zuul.gramineproject.io/t/scaffolding/build/898d3fc3198a4aaebbd84cc7a1c2210f) SUCCESS in 10m 45s
:heavy_check_mark: [test-java-jar ](https://zuul.gramineproject.io/t/scaffolding/build/37cde2fe8d0b4d1bab2a08905f042bde) SUCCESS in 6m 52s
:heavy_check_mark: [test-java-gradle ](https://zuul.gramineproject.io/t/scaffolding/build/20f50b03159745e1bd3b0eaf335cfc49) SUCCESS in 15m 25s
:heavy_check_mark: [test-dotnet ](https://zuul.gramineproject.io/t/scaffolding/build/48b06bdb5c434270b380fecb6adaf376) SUCCESS in 13m 58s

oshogbo committed:
Move building to Docker

Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>

oshogbo committed:
Change nginx port to 8080

Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>

woju committed:
templates/Dockefile: Add macro for RUN apt-get install

Best practice for using apt-get inside Dockerfiles is to apt-get update,
then apt-get install, then clean up caches as a single RUN. Otherwise
you might get into caching problems. This macro will generate correct
command for a given set of packages.

Use this macro to move Builder.depends from Python world to Dockerfiles.

Signed-off-by: Wojtek Porczyk <woju@invisiblethingslab.com>
Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>

gramine-scaffolding commented:
Starting check jobs.

gramine-scaffolding commented:
Build succeeded.
https://zuul.gramineproject.io/t/scaffolding/buildset/8f17f1a8c7094d9c8e26e7fe0893e188

:heavy_check_mark: [bandit ](https://zuul.gramineproject.io/t/scaffolding/build/a4f46933403b4942baac790374750819) SUCCESS in 38s
:heavy_check_mark: [pylint ](https://zuul.gramineproject.io/t/scaffolding/build/4f1289a8062147269b2a0922bc27d35d) SUCCESS in 49s
:heavy_check_mark: [pytest ](https://zuul.gramineproject.io/t/scaffolding/build/61ee47f772f9451691102b967b56f08a) SUCCESS in 40s
:heavy_check_mark: [build-deb ](https://zuul.gramineproject.io/t/scaffolding/build/923049eccbbb4966928c204c3332e09b) SUCCESS in 1m 06s
:heavy_check_mark: [build-docs ](https://zuul.gramineproject.io/t/scaffolding/build/b3500bbc251545d597ec30860a8c1f24) SUCCESS in 37s
:heavy_check_mark: [test-python ](https://zuul.gramineproject.io/t/scaffolding/build/a0c43efc83da45d3a3d15554a08bf81a) SUCCESS in 3m 50s
:heavy_check_mark: [test-flask ](https://zuul.gramineproject.io/t/scaffolding/build/0edd64bee8d54901bfa85ffa6f384b16) SUCCESS in 5m 49s
:heavy_check_mark: [test-nodejs ](https://zuul.gramineproject.io/t/scaffolding/build/543e9e8fa5884d689a02880919d8c9dc) SUCCESS in 5m 23s
:heavy_check_mark: [test-expressjs ](https://zuul.gramineproject.io/t/scaffolding/build/2201e79ef5dc42bbbd8fed26f0715705) SUCCESS in 11m 27s
:heavy_check_mark: [test-koajs ](https://zuul.gramineproject.io/t/scaffolding/build/67199115712343479269f733641ac64e) SUCCESS in 11m 27s
:heavy_check_mark: [test-java-jar ](https://zuul.gramineproject.io/t/scaffolding/build/61f0c39828bd4ca9bf43c91b001c8a85) SUCCESS in 7m 24s
:heavy_check_mark: [test-java-gradle ](https://zuul.gramineproject.io/t/scaffolding/build/e527592146324521a41b834493eac1ba) SUCCESS in 12m 28s
:heavy_check_mark: [test-dotnet ](https://zuul.gramineproject.io/t/scaffolding/build/66f35b26e597499489e1305e61ff27e9) SUCCESS in 10m 27s

gramine-scaffolding commented:
Starting check jobs.

gramine-scaffolding commented:
Build succeeded.
https://zuul.gramineproject.io/t/scaffolding/buildset/08d7492cad234b6bbc1e7a9f2b33a3e0

:heavy_check_mark: [bandit ](https://zuul.gramineproject.io/t/scaffolding/build/8c26b40c0a1e4715a6c9493c784b80ea) SUCCESS in 38s
:heavy_check_mark: [pylint ](https://zuul.gramineproject.io/t/scaffolding/build/621cc28576204c129bca3c00dba18ae6) SUCCESS in 50s
:heavy_check_mark: [pytest ](https://zuul.gramineproject.io/t/scaffolding/build/d19a8ae1f6db4bd7a23fdf1d02ad078e) SUCCESS in 39s
:heavy_check_mark: [build-deb ](https://zuul.gramineproject.io/t/scaffolding/build/d052a6ea8841484a9eee9a15a0ee3a20) SUCCESS in 1m 06s
:heavy_check_mark: [build-docs ](https://zuul.gramineproject.io/t/scaffolding/build/23a3deb533a74e99960d4f082e1962bf) SUCCESS in 38s
:heavy_check_mark: [test-python ](https://zuul.gramineproject.io/t/scaffolding/build/b1139f250c884f6eabf2c5d5b5ebd204) SUCCESS in 4m 25s
:heavy_check_mark: [test-flask ](https://zuul.gramineproject.io/t/scaffolding/build/503515ddcf5c4e94b5c040fc45dcd3ca) SUCCESS in 6m 46s
:heavy_check_mark: [test-nodejs ](https://zuul.gramineproject.io/t/scaffolding/build/f24757c02c664ab2bf202cc51d1937b6) SUCCESS in 5m 49s
:heavy_check_mark: [test-expressjs ](https://zuul.gramineproject.io/t/scaffolding/build/281c6790ba27417db63ffdb6e9c8585c) SUCCESS in 11m 03s
:heavy_check_mark: [test-koajs ](https://zuul.gramineproject.io/t/scaffolding/build/6971c8c1792040f99159826a19e0d3aa) SUCCESS in 12m 34s
:heavy_check_mark: [test-java-jar ](https://zuul.gramineproject.io/t/scaffolding/build/75be418fbc69431396386695989e67eb) SUCCESS in 7m 47s
:heavy_check_mark: [test-java-gradle ](https://zuul.gramineproject.io/t/scaffolding/build/4bec3b41fa60492e9b57cd975b32d32a) SUCCESS in 11m 06s
:heavy_check_mark: [test-dotnet ](https://zuul.gramineproject.io/t/scaffolding/build/8d3e3b47406c45f092ff5f411fa01e37) SUCCESS in 10m 50s

oshogbo committed:
Fix publish ports

The Java Gradle doesn't require one, however, Express.js does.

Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>

gramine-scaffolding commented:
Starting check jobs.

gramine-scaffolding commented:
Build succeeded.
https://zuul.gramineproject.io/t/scaffolding/buildset/e852541aea284b2d8e88a88c43236d9c

:heavy_check_mark: [bandit ](https://zuul.gramineproject.io/t/scaffolding/build/818453e8b50b4606a86fd89f7a628706) SUCCESS in 39s
:heavy_check_mark: [pylint ](https://zuul.gramineproject.io/t/scaffolding/build/518fc89314214f91bae564a892f26972) SUCCESS in 50s
:heavy_check_mark: [pytest ](https://zuul.gramineproject.io/t/scaffolding/build/99b5977d98a24994a2b42e089f75d297) SUCCESS in 38s
:heavy_check_mark: [build-deb ](https://zuul.gramineproject.io/t/scaffolding/build/283c2f88e0e841cbac160efad8e7cca4) SUCCESS in 1m 06s
:heavy_check_mark: [build-docs ](https://zuul.gramineproject.io/t/scaffolding/build/a02017bc81964a25805c5b201596380c) SUCCESS in 39s
:heavy_check_mark: [test-python ](https://zuul.gramineproject.io/t/scaffolding/build/d3d8604fa447441aab81d85f64e039de) SUCCESS in 4m 52s
:heavy_check_mark: [test-flask ](https://zuul.gramineproject.io/t/scaffolding/build/f8edc6322d5f481c907f2679f73cb5e3) SUCCESS in 6m 31s
:heavy_check_mark: [test-nodejs ](https://zuul.gramineproject.io/t/scaffolding/build/910d56892b4e41428ad5e59ebb20931f) SUCCESS in 6m 24s
:heavy_check_mark: [test-expressjs ](https://zuul.gramineproject.io/t/scaffolding/build/d2974933aaad4d81af44c697c82c5408) SUCCESS in 11m 57s
:heavy_check_mark: [test-koajs ](https://zuul.gramineproject.io/t/scaffolding/build/6b05a28e74004cc78925cec867ba6b21) SUCCESS in 12m 47s
:heavy_check_mark: [test-java-jar ](https://zuul.gramineproject.io/t/scaffolding/build/4284195fe5014e98a0bfefacd2c83ab3) SUCCESS in 9m 17s
:heavy_check_mark: [test-java-gradle ](https://zuul.gramineproject.io/t/scaffolding/build/c8998f08bb62474f8967c3203a8891aa) SUCCESS in 13m 11s
:heavy_check_mark: [test-dotnet ](https://zuul.gramineproject.io/t/scaffolding/build/e1e5bdb66cc3489287cf717730b5a4f6) SUCCESS in 13m 31s

bsulich2 reviewed (approved):
> Reviewed 8 of 19 files at r1, 2 of 19 files at r3, 18 of 47 files at r4, 14 of 21 files at r7, 2 of 2 files at r8, 3 of 11 files at r9, 3 of 6 files at r12, 7 of 16 files at r13, 7 of 20 files at r14, 2 of 2 files at r15, 3 of 9 files at r17, 3 of 7 files at r20, 2 of 2 files at r21, 2 of 4 files at r22, 7 of 9 files at r23, 1 of 2 files at r24, 4 of 4 files at r25, 14 of 25 files at r26, 10 of 11 files at r27, 4 of 4 files at r28, 1 of 1 files at r29, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/intel/ScaffoldingForGramine/22)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @woju)



<!-- Sent from Reviewable.io -->

oshogbo reviewed (approved):
> Reviewed 8 of 19 files at r1, 2 of 19 files at r3, 18 of 47 files at r4, 14 of 21 files at r7, 2 of 2 files at r8, 3 of 11 files at r9, 3 of 6 files at r12, 7 of 16 files at r13, 7 of 20 files at r14, 2 of 2 files at r15, 3 of 9 files at r17, 3 of 7 files at r20, 2 of 2 files at r21, 2 of 4 files at r22, 7 of 9 files at r23, 1 of 2 files at r24, 4 of 4 files at r25, 14 of 25 files at r26, 10 of 11 files at r27, 4 of 4 files at r28, 1 of 1 files at r29, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/intel/ScaffoldingForGramine/22)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @woju)



<!-- Sent from Reviewable.io -->
